src/core/server/http/http_server.ts

```diff
@@ -61,20 +65,36 @@ export class HttpServer {
     return this.server !== undefined && this.server.listener.listening;
   }
 
-  public registerRouter(router: Router) {
+  private registerRouter(router: Router) {
     if (this.isListening()) {
       throw new Error('Routers can be registered only when HTTP server is stopped.');
     }
 
+    this.log.debug(`registering route handler for [${router.path}]`);
     this.registeredRouters.add(router);
   }
 
-  public async start(config: HttpConfig): Promise<HttpServerInfo> {
-    this.log.debug('starting http server');
-
+  public setup(config: HttpConfig): HttpServerSetup {
     const serverOptions = getServerOptions(config);
     this.server = createServer(serverOptions);
 
+    return {
+      options: serverOptions,
```

Comment:
`deepFreeze` might be a good idea here

Reply:
aah yeah. I wanted to add freeze here, but found that we have 2 implementations - one for js in `x-pack/secuirty`, one for ts in `core/public/utils`. I think it makes sense to reach an agreement how we going to distribute environment agnostic code snippets, case with `deepFreeze` is perfect example. Do we have such agreement? If not I will create a discussion

Reply:
I do not think that we do have a guideline on sharing code other than things that live as `@kbn/` packages. Could make sense to move this utility there, but it's so small, I'm not sure the boilerplate of a separate package is worth it.

Reply:
turned out the correct implementation is not so simple https://github.com/Microsoft/TypeScript/issues/13923

for now I will copy this functionality, but we need to extract it in a separate package later